server/src/test/java/org/elasticsearch/common/util/ArrayUtilsTests.java

```diff
@@ -30,15 +30,15 @@
 public class ArrayUtilsTests extends ESTestCase {
     public void testBinarySearch() throws Exception {
         for (int j = 0; j < 100; j++) {
-            int index = Math.min(randomInt(0, 10), 9);
-            double tolerance = Math.random() * 0.01;
-            double lookForValue = randomFreq(0.9) ? -1 : Double.NaN; // sometimes we'll look for NaN
+            int index = randomIntBetween(0, 9);
+            double tolerance = randomDoubleBetween(0, 0.01, true);
+            double lookForValue = frequently() ? -1 : Double.NaN; // sometimes we'll look for NaN
             double[] array = new double[10];
             for (int i = 0; i < array.length; i++) {
                 double value;
-                if (randomFreq(0.9)) {
-                    value = Math.random() * 10;
-                    array[i] = value + ((randomFreq(0.5) ? 1 : -1) * Math.random() * tolerance);
+                if (frequently()) {
+                    value = randomIntBetween(0, 10);
```

Comment:
To match the old behavior I think this should be `randomDoubleBetween(0, 10, true)`.

Reply:
Thanks, I missed this being a `double`. Upon further inspection I now think it should be `randomDoubleBetween(0, 9, true)` but including the edge values of the interval shouldn't really matter here.